src/components/common/AppButton.vue

```diff
@@ -30,12 +30,37 @@ export default defineComponent({
     to: {
       type: String,
       required: false,
-    }
+    },
+    variant: {
+      type: String,
+      default: 'link-button',
+      validator(value) {
+        return [
+          'filter-button',
+          'form-button',
+          'internal-link',
+          'external-link'
+        ].includes(value);
+      },
```

Comment:
So, actually I think we could get rid of this prop.
Why? Well, in deciding the component we use in the `componentVariant` computed - we can only check for presence of very specific props/attributes being passed in - such as prop `to` - which means it must be a router-link or attribute `href` - which means it must be an `a` tag.
Otherwise it's just a button 😄
Having the requirement to also need to pass the variant makes our component a bit clunkier to use - if I want an `a` tag button I need to pass both `href` and appropriate `variant` prop.

Reply:
Good point! The simpler the better!